src/components/editor-content/editor-content.tsx

```diff
@@ -25,23 +31,206 @@ interface Props {
+      });
+
+      const startMoment = moment().year(selectedStartDate.getFullYear()).week(startWeek);
+      const endMoment = startMoment.clone();
+      selectedEndDate && endMoment.year(selectedEndDate.getFullYear());
+      endWeek && endMoment.week(endWeek);
+
+      const filteredWeekEntries = weekEntries.filter(
+        entry => TimeUtils.DateInRange(
+          startMoment.startOf("week"),
+          endMoment.endOf("week"),
+          TimeUtils.getWeekFromEntry(entry)
+        )
+      );
+
+      filteredWeekEntries.sort(TimeUtils.sortEntriesByWeek);
+
+      const { workTimeData, workTimeTotalData } = WorkTimeDataUtils.weeksYearsAndMonthsPreprocess(filteredWeekEntries, FilterScopes.WEEK);
+
+      setDisplayedTimeData(workTimeData);
+      setDisplayedTotal(workTimeTotalData);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  /**
+   * Load the month data
+   */
+  const loadMonthData = async () => {
+    if (!person || !selectedStartDate) {
+      return;
+    }
+
+    try {
+      const monthEntries = await Api.getTimeBankApi().timebankControllerGetTotal({
+        personId: person.id.toString(),
+        retention: TimebankControllerGetTotalRetentionEnum.MONTH
+      });
+
+      const startMoment = moment().year(selectedStartDate.getFullYear()).month(selectedStartDate.getMonth());
+      const endMoment = startMoment.clone();
+      selectedEndDate && endMoment.year(selectedEndDate.getFullYear()).month(selectedEndDate.getMonth());
+
+      const filteredMonthEntries = monthEntries.filter(
+        entry => TimeUtils.DateInRange(
+          startMoment.startOf("month"),
+          endMoment.endOf("month"),
+          TimeUtils.getMonthFromEntry(entry)
+        )
+      );
+
+      filteredMonthEntries.sort(TimeUtils.sortEntriesByMonth);
+
+      const { workTimeData, workTimeTotalData } = WorkTimeDataUtils.weeksYearsAndMonthsPreprocess(filteredMonthEntries, FilterScopes.MONTH);
+  
+      setDisplayedTimeData(workTimeData);
+      setDisplayedTotal(workTimeTotalData);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+  
+  /**
+  * Load the year data
+  */
+  const loadYearData = async () => {
+    if (!person || !selectedStartDate) {
+      return;
+    }
+
+    try {
+      const yearEntries = await Api.getTimeBankApi().timebankControllerGetTotal({
+        personId: person.id.toString(),
+        retention: TimebankControllerGetTotalRetentionEnum.YEAR
+      });
+
+      const startMoment = moment().year(selectedStartDate.getFullYear());
+      const endMoment = startMoment.clone();
+      selectedEndDate && endMoment.year(selectedEndDate.getFullYear());
+
+      const filteredYearEntries = yearEntries.filter(
+        entry => TimeUtils.DateInRange(
+          startMoment.startOf("year"),
+          endMoment.endOf("year"),
+          TimeUtils.getYearFromEntry(entry)
+        )
+      );
+  
+      yearEntries.sort((year1, year2) => year1.id?.year! - year2.id?.year!);
```

Comment:
Move to TimeUtils like in filteredMonthEntries.sort(TimeUtils.sortEntriesByMonth) just to keep things consistent